includes/component.php

```diff
@@ -767,7 +767,7 @@ public function delete_doc_activity( $new_status, $old_status, $post ) {
 	 * @return str $link The filtered permalink
 	 */
 	function filter_permalinks( $link, $post, $leavename, $sample ) {
-		if ( bp_docs_get_post_type_name() == $post->post_type ) {
+		if ( bp_docs_get_post_type_name() == $post->post_type && ! empty( $post->post_name ) ) {
```

Comment:
Why this change? Don't pending items have `post_name`?

Reply:
This was necessary when I was using the WP built-in pending status, which doesn't allow a `post_name` to be generated. There were other problems, too, so I ended up using a custom status (that was your suggestion, and it was the right move--way easier than handling all the odd behavior of the built-in pending status).